File: exp-10/q4-adjacency-list/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int n, m;
    cin >> n >> m;
    vector<vector<int>> adj(n + 1);
    for (int i = 0; i < m; i++)
    {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    for (int i = 1; i <= n; i++)
        sort(adj[i].begin(), adj[i].end());
    for (int i = 1; i <= n; i++)
    {
        cout << i << ": ";
        for (int j = 0; j < (int)adj[i].size(); j++)
        {
            if (j > 0)
                cout << " ";
            cout << adj[i][j];
        }
        cout << endl;
    }
    return 0;
}

File: exp-10/q4-adjacency-list/solution.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let lines = [];
rl.on('line', line => lines.push(line.trim()));
rl.on('close', () => {
    const data = lines.join(' ').split(/\s+/).map(Number);
    let idx = 0;
    const n = data[idx++], m = data[idx++];
    const adj = Array.from({ length: n + 1 }, () => []);
    for (let i = 0; i < m; i++) {
        const u = data[idx++], v = data[idx++];
        adj[u].push(v);
        adj[v].push(u);
    }
    for (let i = 1; i <= n; i++) adj[i].sort((a, b) => a - b);
    for (let i = 1; i <= n; i++) {
        console.log(i + ': ' + adj[i].join(' '));
    }
});